manifests/osfamily/suse.pp

```diff
@@ -33,11 +34,11 @@
 
       # Given the path to a key, see if it is imported, if not, import it
       exec {  "import-${keyname}":
-        path      => '/bin:/usr/bin:/sbin:/usr/sbin',
-        command   => "rpm --import ${gpg_path}",
-        unless    => "rpm -q gpg-pubkey-`echo $(gpg --throw-keyids < ${gpg_path}) | cut --characters=11-18 | tr [A-Z] [a-z]`",
-        require   => File[$gpg_path],
-        logoutput => 'on_failure',
+        path           => '/bin:/usr/bin:/sbin:/usr/sbin',
+        command        => "rpm --import ${gpg_path}",
+        unless         => "rpm -q gpg-pubkey-$(echo $(gpg --homedir ${gpg_homedir} --throw-keyids < ${gpg_path}) | cut --characters=11-18 | tr [A-Z] [a-z])",
```

Comment:
I'm trying to come up with less painful variations to this line and not coming up with much.

There is a minimal change that I think would be helpful:

rpm -q gpg-pubkey-$(gpg --homedir ${gpg_homedir} --throw-keyids --with-colons < ${gpg_path} | cut -d: -f5 | tr [A-Z] [a-z])

Should at least get away from specifying the character positions (which I think were off by one).  Also I /think/ the echo was unnecessary?